fischkapp/src/components/CardList.tsx

```diff
@@ -1,13 +1,38 @@
 import { flashCardsData } from "../data/data";
 import { AddNewCard } from "./AddNewCard";
 import { StyledCardList } from "./styles/CardList.styled";
-import { FrontCard } from "./CreatedCard.tsx/FrontCard";
 import { CreatedCard } from "./CreatedCrad";
+import { useState, useEffect } from "react";
 
-export const CardList = () => {
+interface FlashCardType {
+  id?: number;
+  frontText: String;
+  backText: String;
+}
+
+type CardListPropTypes = {
+  onIsNewCard: boolean;
+  onSetNewCard: Function;
+  onFlashCards: FlashCardType[];
+  onSetFlashCards: Function;
+};
+
+export const CardList = ({
+  onIsNewCard,
+  onSetNewCard,
+  onFlashCards,
+  onSetFlashCards,
+}: CardListPropTypes) => {
   return (
     <StyledCardList>
-      {flashCardsData.map((flashCard) => (
+      {onIsNewCard ? (
+        <AddNewCard
+          onSetNewCard={onSetNewCard}
+          onFlashCards={onFlashCards}
+          onSetFlashCards={onSetFlashCards}
+        />
+      ) : null}
+      {onFlashCards.map((flashCard) => (
```

Comment:
**nitpick:** `onFlashCards` is named incorrectly. Every `on` prefix means that this prop is a function. When you name data, you should name it more precisely, so it's clear what it is. In this case it would be enough to name it `flashcards`